proc_scripts/align_slice.py

```diff
@@ -27,4 +27,33 @@ def align_and_slice(s, dimension='t2', convwidth=500, threshold=0.05, fl=None):
     print("slicing to",slices)
     if fl is not None:
         fl.pop_marker()
-    return s[dimension:slices]
+    return s[axis:slices]
+def correlation_align(s,avg,convwidth=0,axis='t2',fl=None):
+    assert not s.get_ft_prop(axis), 'I want time-domain data'
+    assert not avg.get_ft_prop(axis), 'I want time-domain data'
+    c = avg.run(conj)
+    R = pi*convwidth # Lorentzian FWHM formula -- R = 1/T2, FWHM = R/pi
+    # ALSO convolve both
+    check = avg * c * exp(-2*R*abs(s.fromaxis(axis)))
+    assert s.get_ft_prop(axis,['start','freq']) is not None, "should be FT'd first, so the startpoint is set"
+    check.ft(axis, pad=2**14)
+    check.reorder(axis,first=False)
+    forplot = abs(check)[axis:(-50,50)]
+    myline = forplot.C.argmax(axis)
+    indirect_dims = set(s.dimlabels) - set([axis])
+    phcyc_dims = [j for j in indirect_dims if j.startswith('ph')]
+    phcyc_dims.sort()
+    indirect_dims = list(set(indirect_dims) - set(phcyc_dims))
+    #forplot.reorder(phcyc_dims+indirect_dims)
+    forplot.smoosh(phcyc_dims+indirect_dims,dimname='indirect',noaxis=True).setaxis('indirect','#').reorder('indirect',first=False)
+    if fl is not None:
+        fl.push_marker()
+        fl.next('cross-correlation')
+        fl.image(forplot,human_units=False)
```

Comment:
running into issues here, 7/8/20

Reply:
okay so I did some comparisons with previous versions on the git tree that I know produced the plot at least. proc_echo_mw now AT LEAST can produce the plot and we can begin troubleshooting to see what's wrong with it/what adjustments need to be made. Take a look and let me know any thoughts or concerns you may have!

Reply:
so, in group meeting, why don't you present the two alignment methods -- `align_slice` and the correlation alignment, and explain the math/science ideas behind them.  As I'm saying elsewhere, I think the main bottleneck here is understanding/ideas-based, not a problem with the coding.